react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js

```diff
@@ -9,15 +9,26 @@ const initialState = {
 function reducer(state = initialState, action) {
   switch (action.type) {
     case actions.GET_BOOKS: // TODO to implement the logic
-      return { ...state };
+      return { ...state, originalData: action.payload.data, books: action.payload.data };
     case actions.ADD_TO_CART: // TODO to implement the logic
-      return { ...state };
+      return { ...state, bookSelected: [...state.bookSelected, action.payload] };
     case actions.ADD_ITEM: // TODO to implement the logic
-      return { ...state };
+      return {
+        ...state,
+        bookSelected: state.bookSelected.reduce((acum, curr) => {
+          curr = curr.id === action.payload ? { ...curr, quantity: curr.quantity + 1 } : curr;
+          return [...acum, curr];
```

Comment:
One last thing, you can avoid the `return` and the `curr = ..` removing the braces and using the implicit return of the arrow functions.
```js
bookSelected.reduce((acum, curr) =>
  [ ...acum, curr.id === action.payload ? { ...curr, quantity: curr.quantity + 1 } : curr ]
  , [])
```